src/fetching/RelayConnectionFetcher.js

```diff
@@ -72,22 +72,17 @@ export default class RelayConnection<T, Data> implements Fetcher<Data[]> {
 		};
 
 		while (retries < this.maxRetries) {
-			const res = await fetch(url, options);
-			const data = await res.json();
+			try {
+				const res = await fetch(url, options);
+				const data = await res.json();
 
-			if (data.errors && Array.isArray(data.errors)) {
+				return getConnection(data);
+			} catch (error) {
```

Comment:
I don't think this will catch errors like 429s or 500s, right?

Reply:
... surely not

Reply:
I think it should. What do you think of this:

```javascript
while (retries < this.maxRetries) {
    try {
       const res = await fetch(url, options);
       const data = await res.json();
        // retry on TOO MANY REQUESTS and server errors
        if (res.status === 429 || res.status > 499) {
            retries = retries + 1;
        } else {
            if (data.errors && Array.isArray(data.errors) && handleErrors) {
                handleErrors(data);
            }
            return getConnection(data);
        }
    } catch(error) {
        console.log("RelayConnectionFetcher: Received errors while fetching connection: ", error);
		retries = retries + 1;
    }
}
```